cli/src/commands/workspace/add.rs

```diff
@@ -58,6 +59,10 @@ pub struct WorkspaceAddArgs {
     /// new r1 r2 r3 ...`.
     #[arg(long, short)]
     revision: Vec<RevisionArg>,
+    /// Do not copy sparse patterns from the current workspace into the new
+    /// workspace.
+    #[arg(long)]
+    no_sparse_patterns: bool,
```

Comment:
I would actually guess that this flag would give you an empty list of sparse patterns. Maybe we want to make it `--sparse-patterns=copy/full/empty` or something like that?

Reply:
Good idea. I've got a patch incoming that will do all this.